### PR TITLE
COR-676: form/upload validation

### PR DESCRIPTION
**Pull request by internetti**
some of the select fields were still accepting random values on upload
booleans were not checked for valid values on upload
birthdays were not checked on minimum valid value on upload

added the missing validations and unified the rest

### CODE REVIEW

File: internal/api/enumTypes/country.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package api
+package enumTypes
```
Comment (neb42) [inline internal/api/enumTypes/country.go:1]: Why has country moved?

File: internal/api/enumTypes/disability_level.go [hunk 2]
```diff
@@ -49,14 +49,24 @@ func ParseDisabilityLevel(str string) (DisabilityLevel, error) {
 	switch str {
 	case string(DisabilityLevelNone):
 		return DisabilityLevelNone, nil
+	case DisabilityLevelNone.String():
+		return DisabilityLevelNone, nil
```
Comment (neb42) [inline internal/api/enumTypes/disability_level.go:53]:
```
case string(DisabilityLevelNone), DisabilityLevelNone.String():
  return DisabilityLevelNone, nil
```

File: internal/api/enumTypes/country.go (hunk 1 above)
Comment (internetti) [reply internal/api/enumTypes/country.go:1]: I classified it as a list of options for the form, but you're right, doesn't belong here, I'll move it back

**Comment by internetti** [comment]
@neb42 the enumTypes folder is not necessary per se, I just felt that these types do play a different role than the other ones in that folder, so I wanted to group them together to find them more easily

**Pull request merged**